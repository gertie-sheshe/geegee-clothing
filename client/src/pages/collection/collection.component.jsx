import React from 'react';
import { connect } from 'react-redux';

import CollectionItem from '../../components/collection-item/collection-item.component';

import {
  CollectionPageContainer,
  CollectionTitleContainer,
  CollectionItemsContainer,
} from './collection.styles';

// Selectors
import { selectCollection } from '../../redux/shop/shop.selectors';

const CollectionPage = ({ collection }) => {
  const { title, items } = collection;

  return (
    <CollectionPageContainer>
      <CollectionTitleContainer>{title}</CollectionTitleContainer>
      <CollectionItemsContainer>
        {items.map(item => (
          <CollectionItem item={item} key={item.id} />
        ))}
      </CollectionItemsContainer>
    </CollectionPageContainer>
  );
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state),
});

export default connect(mapStateToProps)(CollectionPage);
