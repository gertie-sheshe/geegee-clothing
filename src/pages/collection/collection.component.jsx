import React from 'react';
import {connect} from 'react-redux';

import CollectionItem from '../../components/collection-item/collection-item.component';

import './collection.styles.scss';

// Selectors
import {selectCollection} from '../../redux/shop/shop.selectors'

const CollectionPage = ({ match, collection }) => {
  console.log('COLLECTION', collection)
  return (<div className="collection-page">
    <h2>CATEGORY PAGE</h2>
    
  </div>)
};

const mapStateToProps = (state, ownProps) => ({
    collection: selectCollection(ownProps.match.params.collectionId)(state)
});

export default connect(mapStateToProps)(CollectionPage);
