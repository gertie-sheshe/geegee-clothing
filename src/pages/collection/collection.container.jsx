import { connect } from 'react-redux';
import { compose } from 'redux';

import CollectionPage from './collection.component';

import WithSpinner from '../../components/with-spinner/with-spinner.component';

import { createStructuredSelector } from 'reselect';
import { selectIsCollectionLoaded } from '../../redux/shop/shop.selectors';

const mapStateToProps = createStructuredSelector({
  isLoading: state => !selectIsCollectionLoaded(state),
});

const CollectionPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner,
)(CollectionPage);

export default CollectionPageContainer;
