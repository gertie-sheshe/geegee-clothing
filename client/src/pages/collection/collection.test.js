import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './collection.component';
import CollectionItem from '../../components/collection-item/collection-item.component';

describe('CollectionPage Tests', () => {
  let wrapper;
  let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    const mockCollection = {
      items: mockItems,
      title: 'Test',
    };

    wrapper = shallow(<CollectionPage collection={mockCollection} />);
  });

  it('Should render the CollectionPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the same number of CollectionItems as collection array', () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  });
});
