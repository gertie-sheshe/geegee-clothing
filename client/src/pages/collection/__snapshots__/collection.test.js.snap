// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage Tests Should render the CollectionPage Component 1`] = `ShallowWrapper {}`;
